### PR TITLE
cli: install helm charts in cli instead of bootstrapper

### PR DESCRIPTION
**Pull request by elchead**
### Context

Part of the goal to move towards `constellation apply` and brings us closer to manage helm charts through terraform.

### Proposed change(s)

- move all charts installation except Cilium to CLI

- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [ ] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

**Comment by elchead** [comment]
testing GCP: https://github.com/edgelesssys/constellation/actions/runs/5666731041

**Comment by elchead** [comment]
testing on azure: https://github.com/edgelesssys/constellation/actions/runs/5668918594

File: internal/cloud/azureshared/appcredentials.go [hunk 1]
```diff
@@ -37,8 +40,16 @@ func ApplicationCredentialsFromURI(cloudServiceAccountURI string) (ApplicationCr
 		return ApplicationCredentials{}, fmt.Errorf("invalid service account URI: invalid host: %s", uri.Host)
 	}
 	query := uri.Query()
+
+	subscriptionPattern := regexp.MustCompile(`subscriptions/([^/]+)/`)
```
Comment (3u13r) [inline internal/cloud/azureshared/appcredentials.go:44]: `regexp.MustCompile` always go to the top of the file as `var`.

File: internal/cloud/azureshared/appcredentials.go [hunk 2]
```diff
@@ -37,8 +40,16 @@ func ApplicationCredentialsFromURI(cloudServiceAccountURI string) (ApplicationCr
...
+	subscriptionID := getFirstMatchOrEmpty(subscriptionPattern, query.Get("uami_resource_id"))
+	rgPattern := regexp.MustCompile(`resourceGroups/([^/]+)/`)
```
Comment (3u13r) [inline internal/cloud/azureshared/appcredentials.go:46]: regexp.MustCompile always go to the top of the file as var.

**Comment by derpsteb** [comment]
AWS: https://github.com/edgelesssys/constellation/actions/runs/5677935261

**Review by daniel-weisse — DISMISSED** [review]
We should split [AB#3294](https://dev.azure.com/Edgeless/Edgeless/_workitems/edit/3294/) into more individual tickets as I feel this PR currently does not achieve what the ticket was meant to do.
While with this PR, we install most of the charts on the CLI side, Cilium is still handled by the bootstrapper (which I believe can already be achieved using just the CLI with some workarounds. Might not be worth the implementation effort though, depending on how soon we plan on upgrading Cilium).
The "new" helm install doesn't share any logic with the upgrade logic, nor did we achieve in loading values in just one place.

I'm Ok with merging this PR in its current feature state, but we should not close the Ticket, rather split it into at least 3 (move helm install logic to CLI, unify values loading, unify helm install and upgrade logic).

**Comment by elchead** [comment]
> We should split [AB#3294](https://dev.azure.com/Edgeless/Edgeless/_workitems/edit/3294/) into more individual tickets as I feel this PR currently does not achieve what the ticket was meant to do. While with this PR, we install most of the charts on the CLI side, Cilium is still handled by the bootstrapper (which I believe can already be achieved using just the CLI with some workarounds. Might not be worth the implementation effort though, depending on how soon we plan on upgrading Cilium). The "new" helm install doesn't share any logic with the upgrade logic, nor did we achieve in loading values in just one place.
> 
> I'm Ok with merging this PR in its current feature state, but we should not close the Ticket, rather split it into at least 3 (move helm install logic to CLI, unify values loading, unify helm install and upgrade logic).

I agree that several PRs more are necessary and I created new Azure subtickets like you suggested

File: cli/internal/cmd/init.go [hunk 3]
```diff
@@ -214,10 +228,22 @@ func (i *initCmd) initialize(cmd *cobra.Command, newDialer func(validator atls.V
...
+	bufferedOutput := &bytes.Buffer{}
+	err = i.writeOutput(idFile, resp, flags.mergeConfigs, bufferedOutput, fileHandler)
+	if err != nil {
+		return err
+	}
+	if i.helmInstaller != nil {
+		if err := i.helmInstaller.Install(cmd.Context(), provider, masterSecret, idFile, serviceAccURI, releases); err != nil {
+			return fmt.Errorf("installing Helm charts: %w", err)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (elchead) [reply cli/internal/cmd/init.go:244]: it's just a dirty trick to simplify testing. Should I instead inject a test dummy?

File: internal/cloud/azureshared/appcredentials.go (hunk 1 above)
Comment (elchead) [reply internal/cloud/azureshared/appcredentials.go:44]: is there a reason?

**Comment by elchead** [comment]
running E2E again: https://github.com/edgelesssys/constellation/actions/runs/5679687285

**Comment by elchead** [comment]
can you have another look and approve? @daniel-weisse @derpsteb

File: cli/internal/cmd/init_test.go [hunk 4]
```diff
@@ -666,3 +667,12 @@ func (c stubInitClient) Recv() (*initproto.InitResponse, error) {
 
 	return res, err
 }
+
+type placeholderHelmInstaller struct{}
```
Comment (derpsteb) [inline cli/internal/cmd/init_test.go:671]: These are called stub* in other places.

**Review by derpsteb — APPROVED** [review]

**Pull request merged**